content/docs/sync-async-boundary.md

```diff
@@ -0,0 +1,43 @@
+
+The main problem with the synchronous approach is the temporal coupling it introduces. In order to process the request all the components involved need to be available at the same time. If processing the initial request results in more request (which in turn generate more requests) it can quickly get out of hand. The more parties need to be available at the same time, the higher the likelihood of any one of them failing and undermining the whole process.
+
+The asynchronous approach solves this problem by introducing queues and one-way messages. The sender or publisher puts the message onto a queue or topic and can immediately move on to processing subseqent work items. It does not waste time and resources actively waiting for a response. If the recipient is unavailable that OK too. The queue will ensure the message is not lost and that it will be eventually delivered. 
+
+This huge advantage of asynchronous systems becomes a disadvantage when it comes to building user interface components. The time to process a message is generally longer and more varied than in a synchronous system. In order for the user to know the state of the system, it might be necessary to visualize pending operations related to messages that have been sent but not yet processed. This introduces the need to keep the state on the client side. Depending on the complexity of the problem, it can lead to the need to implement fully-fledged multi-master replication as multiple clients can send messages that compete to modify the same set of data.
+
+## The middle ground
+
+Whenever there are two extreme approaches, each with their strong and weak points, someone has to make a decision to pick one or the other. But what if such a decision did not have to be made for the entire system? Here's where the sync-async boundary becomes useful.
+
+The synchronous approach shines when used close to the user and for work that is highly interactive, like adding or removing items from a shopping cart. It begins to cause problems when used to implement complex logic or long-running non-interactive processes. Fortunately this is precisely where the asynchronous approach works best. 
+
+## The pattern
+
+Divide the system in two parts. The synchronous part includes components running in the browser and ones that process HTTP requests sent by the browser. The asynchronous part contains all the components which do not interact with the user. Let's call the former *edge* and the latter *interior*.
+
+The user interacts with the *edge* of the system by working on a private set of data in a synchronous manner. Once that data set is ready, it can be submitted for processing. This is when, at the boundary, the sunmit HTTP request is transformed into a message that captures the user-created data and notifies the components in the *interior* of the system.
```

Comment:
```suggestion
The user interacts with the *edge* of the system by working on a private set of data in a synchronous manner. Once that data set is ready, it can be submitted for processing. This is when, at the boundary, the submit HTTP request is transformed into a message that captures the user-created data and notifies the components in the *interior* of the system.